### DISCUSSION TITLE
About naming because of withPoints. For version 4.0

### DISCUSSION THREAD

**Discussion by cvvergara** [pgRouting/pgrouting#2225] [Ideas]
Currently there are the following proposed functions and they all work with dijkstra

* pgr_withPoints
* pgr_withPointsCost
* pgr_withPointsCostMatrix
* pgr_withPointsKSP
* pgr_withPointsDD
* pgr_withPointsVia -- to be done

Thinking about trsp (for when they are developed):

* pgr_trsp_withPoints
* pgr_trsp_withPointsCost
* pgr_trsp_withPointsCostMatrix
* pgr_trsp_withPointsKSP
* pgr_trsp_withPointsDD
* pgr_trsp_withPointsVia

Thinking about aStar (for when they are developed):

* pgr_aStar_withPoints
* pgr_aStar_withPointsCost
* pgr_aStar_withPointsCostMatrix
* pgr_aStar_withPointsKSP
* pgr_aStar_withPointsDD
* pgr_aStar_withPointsVia

Having this convention when with points are involved, makes it easier to remember, aka less complicated API

**Comment by robe2** [comment]
Looking at this closer, I think all existing functions only have one _

So should be: pgr_trspWithPoints

and so on.

**Reply by cvvergara** [reply]
Well, it still the first time we encounter this naming problem, and I added the  `_` for readability

Option A (examples in first comment)
```
pgr_<family>_withPoints<termination>
```
Option B
```
pgr_<family>withPoints<termination>
```
examples:
*  pgr_aStarWithPoints
*  pgr_aStarWithPointsCost
*   pgr_aStarWithPointsCostMatrix
*   pgr_aStarWithPointsKSP
*   pgr_aStarWithPointsDD
*   pgr_aStarWithPointsVia

**Reply by robe2** [reply]
You have functions like:

- pgr_aStarCost
- pgr_aStarCostMatrix
- pgr_primDD
- pgr_kruskalDD

so I feel this is just an extension of that naming

they aren't 

- pgr_aStar_Cost

But I do agree that B is a little clearer but confusing as some would be _aStar_ vs. aStar_

**Comment by dkastl** [comment]
I actually like using `_` for readability. And I'm supporting conventions. Let me post here the answer I gave to Vicky in another channel:

Regarding naming conventions, we could say that "left -> right" should indicate "general -> more specific".

For example:

- `pgr` indicates this is pgRouting, the most general indicator
- `astar` indicates the algorithm
- `withPoints` is something more specific as well as `via`. But I don't know which is more or less specific, if one requires the other or if they interchangeable.

About the `via` (or other endings):

if `Via` makes only sense with `withPoints`, then I would make it the last part of the name. 
If any order works, then I feel `withPointsVia` is easier to speak
